src/main/java/org/opensearch/flowframework/workflow/GetMLTaskStep.java

```diff
@@ -95,4 +83,46 @@ public String getName() {
         return NAME;
     }
 
+    /**
+     * Retryable GetMLTask
+     * @param workflowId the workflow id
+     * @param getMLTaskFuture the workflow step future
+     * @param taskId the ml task id
+     * @param retries the current number of request retries
+     */
+    protected void retryableGetMlTask(String workflowId, CompletableFuture<WorkflowData> getMLTaskFuture, String taskId, int retries) {
+        mlClient.getTask(taskId, ActionListener.wrap(response -> {
+            if (response.getState() != MLTaskState.COMPLETED) {
+                throw new IllegalStateException("MLTask is not yet completed");
+            } else {
+                logger.info("ML Task retrieval successful");
+                getMLTaskFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry(MODEL_ID, response.getModelId()),
+                            Map.entry(REGISTER_MODEL_STATUS, response.getState().name())
+                        ),
+                        workflowId
+                    )
+                );
+            }
+        }, exception -> {
+            if (retries < maxRetry) {
+                // Sleep thread prior to retrying request
+                try {
+                    Thread.sleep(5000);
+                } catch (Exception e) {
```

Comment:
Pretty sure the only exception you're going to get here is `InterruptedException` in which case we probably want to cancel the future?

Reply:
Thats a good point

Reply:
Attempting to cancel the future results in 
```
Forbidden method invocation: java.util.concurrent.Future#cancel(boolean) [Don't interrupt threads use FutureUtils#cancel(Future<T>) instead]
  in org.opensearch.flowframework.workflow.GetMLTaskStep (GetMLTaskStep.java:115)
Scanned 68 class file(s) for forbidden API invocations (in 0.41s), 1 error(s).
```

I'll opt to keep it as is, completing exceptionally on exception.